content/blog/2015/2015-05-12-praktikum-bei-code-for-germany.markdown

```diff
@@ -1,6 +1,6 @@
 ---
 author: Fiona Krakenbuerger
-date: 2015-05-12 14:25
```

Comment:
Looks like we're losing some time information here and in some other markdown files. Don't quite understand why that happens though from the script as it should convert to a `publishedDate` when it finds a `date`.

Reply:
Ah sorry, I forgot to comment on this. I removed the time for some entries manually (for those files were not touched at all since they had no `published` field set at all and were skipped by the script). With the time still present, hugo complained that it could not parse it into a date.

Reply:
It seems like we never displayed the time anyway though 🧐

Reply:
I think it complains about the missing seconds. Here's an updated script that seems to fix it:

```python
import os
from datetime import datetime

import frontmatter


def process_markdown_files(directory):
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith(".md") or file.endswith(".markdown"):
                file_path = os.path.join(root, file)
                with open(file_path, "r", encoding="utf-8") as f:
                    post = frontmatter.load(f)

                published = post.get("published")
                if published is not None:
                    if not published:
                        post["draft"] = "true"
                        del post["published"]
                    else:
                        del post["published"]

                if post.get("date"):
                    date = post.get("date")
                    if isinstance(date, str):
                        date = datetime.fromisoformat(date + ":00")
                    if published:
                        post["publishedDate"] = date
                        del post["date"]
                    else:
                        post["date"] = date

                with open(file_path, "w", encoding="utf-8") as f:
                    f.write(frontmatter.dumps(post, sort_keys=False, width=500))


if __name__ == "__main__":
    process_markdown_files("./content")
```